import_export/widgets.py

```diff
@@ -472,3 +475,76 @@ def clean(self, value, row=None, **kwargs):
     def render(self, value, obj=None):
         ids = [smart_str(getattr(obj, self.field)) for obj in value.all()]
         return self.separator.join(ids)
+
+class ResourceWidget(ManyToManyWidget):
+    """
+    Widget that fully serializes many to many or foreign key relationships
+    using the related model's resource, with support for deserialization.
+    Due to limitations with non-JSON parsers, only supports the Json format
+    for the serialized sub-resource. 
+    :param model: The model the ManyToMany field refers to (required).
+    :param resource_class: Resource class used to serialize/deserialize
+    the related model(s)
+    :param is_m2m: related field is m2m. set False for ForeignKey fields.
+    """
+    # DESIGN DECISION: Only support JSON for this type of nested datastructure.
+    # This simplifies testing and support to such a degree, and the only real
+    # use case for using this over Dehydrate is to move between Django
+    # environments.
+    __format = JSON()
+
+    def __init__(self, model, resource_class, is_m2m=False, **kwargs):
+        self.model = model
+        # Note that this is instantiated once here.
+        self.resource = resource_class()
+        self.is_m2m = is_m2m
+        super().__init__(model, **kwargs)
+    
+    def clean(self, value, row=None, **kwargs):
+        """
+        Deserialize the incoming data, creating models via
+        the resource specified in self.resource. 
+        """
+
+        # for reasons unknown, the Json parser can't handle
+        # an empty string, so we make a blank dataset.
+
+        if not value:
+            dataset = tablib.Dataset()
+        else:
+            dataset = self.__format.create_dataset(value)
+
+        results = self.resource.import_data(dataset)
```

Comment:
yeah this is connected with why this should be a field. Because I always use transactions this worked fine for me, but without use transactions this might be an issue of creating extraneous stuff if you don't go forward. 

So this really needs to return a tablib dataset or even a set of models and have the field do the saving I think.